soundbay/utils/logging.py

```diff
@@ -105,24 +104,27 @@ def update_predictions(self, pred_tuple):
         self.pred_proba_list.append(torch.softmax(pred_tuple[0].data, 1).cpu().numpy())
         self.label_list += label.cpu().numpy().tolist()
 
-    def calc_metrics(self, epoch):
+    def calc_metrics(self, epoch: int, mode: str = 'train', label_names: List[str] = ('Noise', 'Call')):
         """calculates metrics, saves to tensorboard log & flush prediction list"""
         pred_proba_array = np.concatenate(self.pred_proba_list)
         self.metrics_dict = self.get_metrics_dict(self.label_list, self.pred_list, pred_proba_array)
-        self.log_writer.log({'Accuracy': self.metrics_dict['accuracy']}, step=epoch)
-        self.log_writer.log({'f1score': self.metrics_dict['f1score']}, step=epoch)
-        self.log_writer.log({'precision': self.metrics_dict['precision']}, step=epoch)
-        self.log_writer.log({'recall': self.metrics_dict['recall']}, step=epoch)
-        if not self.debug_mode:
-            self.log_writer.log({'auc': self.metrics_dict['auc']}, step=epoch)
+        self.log_writer.log({f'{mode}_Accuracy': self.metrics_dict['accuracy']}, step=epoch)
+        self.log_writer.log({f'{mode}_f1score': self.metrics_dict['f1score']}, step=epoch)
+        self.log_writer.log({f'{mode}_precision': self.metrics_dict['precision']}, step=epoch)
+        self.log_writer.log({f'{mode}_recall': self.metrics_dict['recall']}, step=epoch)
+        if not self.debug_mode and mode != 'train':
```

Comment:
actually, why not log confusion matrix or pr curve on training as well? Might be an important indicator on the sources of bugs (overfit vs split distribution)